index.ipynb

```diff
@@ -0,0 +1,167 @@
+       "\n",
+       "Explanation:\n",
+       "\n",
+       "* ``singularity exec`` - Ask singularity to execute a command in a container.\n",
+       "* ``-B $XDG_RUNTIME_DIR`` - Bind mount your user specific temporary director. Jupyter uses this directory and\n",
+       "  singularity does not mount it by default.\n",
+       "* ``software.simg`` - The image to launch.\n",
+       "* ``jupyter notebook`` Execute ``jupyter notebook`` inside the image\n",
+       "\n",
+       "Once **jupyter** starts, point your browser to the URL **jupyter** prints on the terminal. **jupyter** inside the\n",
+       "container accesses the configuration in your home directory on the host system. If you have a password configured for\n",
+       "**jupyter** on your host system, use that to login. Otherwise, the URL should include a token that will log you in.\n",
+       "\n",
+       "### Docker container\n",
+       "\n",
+       "The [glotzerlab-software](https://glotzerlab-software.readthedocs.io) image contains all software needed to execute\n",
+       "these notebooks and a copy of the notebooks themselves in ``/fresnel-examples``. Run this command to start **jupyter**:\n",
+       "\n",
+       "```bash\n",
+       "▶ docker run --rm -p 127.0.0.1:9999:9999 glotzerlab/software jupyter notebook --port 9999 --ip 0.0.0.0 --no-browser /fresnel-examples\n",
+       "```\n",
+       "\n",
+       "If you have installed the docker NVIDIA runtime, add ``--runtime=nvidia`` after ``run`` to utilize your GPUs.\n",
+       "\n",
+       "Explanation:\n",
+       "\n",
+       "* ``docker run`` - Ask docker to run a command in a container.\n",
+       "* ``--runtime=nvidia`` - (if applicable) use the NVIDIA runtime to make host GPUS accessible in the container.\n",
+       "* ``--rm`` - Delete the container after exiting.\n",
+       "* ``-p 127.0.0.1:9999:9999`` - Make port 9999 in the container available at ``localhost:9999``.\n",
+       "* ``glotzerlab/software`` - name of the image to execute.\n",
+       "* ``jupyter notebook`` - execute the **jupyter** notebook.\n",
+       "* ``--port 9999 --ip 0.0.0.0`` - **jupyter** should listen on port 9999 for connections from outside the container.\n",
+       "* ``--no-browser`` - Tell Jupyter not to attempt to launch a browser.\n",
+       "* ``/fresnel-examples`` - Location of the example notebooks in the image.\n",
+       "\n",
+       "Once **jupyter** starts, point your browser to ``localhost:9999``. Copy the token from the **jupyter** terminal output\n",
+       "and paste it into the password box to access the notebook.\n"
+      ],
+      "text/plain": [
+       "<IPython.core.display.Markdown object>"
+      ]
+     },
+     "metadata": {},
+     "output_type": "display_data"
+    }
+   ],
+   "source": [
+    "# Run this cell to render the README.md\n",
+    "from IPython.display import display, Markdown\n",
+    "with open('README.md', 'r') as readme:\n",
```

Comment:
Does this code show? Is there a way to hide the code but not the output?

In general, I'm find with the index.ipynb concept. It would be better if mybinder could just display the readme, but if that isn't possible this is OK. It is nice not having to manually replicate information in the readme, though we will need to remember rerun the notebook when updating the readme.

Have you tested if this builds without warning in the fresnel sphinx docs? Sphinx is very particular that every document in the directory tree *must* be included in the output. I've had mixed luck with getting the ignore option to ignore notebooks.

Reply:
I have not tested this with Sphinx and I think I can use tags to hind the code. However, I like the idea of not using an index.ipynb and instead linking people to the notebook they are reading about in the docs. I did a quick look into juypter lab and it looks like you can right click on a markdown file and select preview which will render it, but I haven't found a way (yet) to use url params to trigger that view.